float32[] pos
float32[] vel
float32[] tor
float32[] temp
uint8[] error_code
int8[] pattern